api/types/saml_idp_service_provider.go

```diff
@@ -309,3 +319,12 @@ func (am *SAMLAttributeMapping) CheckAndSetDefaults() error {
 	}
 	return nil
 }
+
+func validatePreset(preset string) bool {
+	switch preset {
+	case samlsp.GCP:
```

Comment:
I think you can also add a `case ""` here and get rid of the `if s.Spec.Preset != ""` check in the calling method altogether.